### ISSUE TITLE
Semi-autoroom settings got clear out after saving the plugin settings

### ISSUE THREAD

**Issue by timothy-morph** [homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#335]
Hi, after a recent update, I've noticed that the semi-autoroom array will disappear after I save the settings for the plug-in.
Tried a few times and it deleted the my semi-autoroom settings after I save.

Could you see if this is a bug?
Thanks :)

And by the way the multi-room cleaning function is awesome!
Thanks for implementing it!

**Comment by normen** [comment]
This is because the variable can be used in multiple ways. Its probably quite hard to implement that in the settings UI. The semi-autoroom will probably always be a "hack-ish" workaround where you have to edit the config file yourself.

Not speaking for the project maintainers though, just giving my view knowing what the issue is here.

**Comment by afharo** [comment]
Yeah, I agree with @normen. `rooms` is supposed to have the full room description. The semi-autoroom feature is an addition to help on the discovery of the IDs, but was never thought to be a long-term solution. Maybe we need to revisit to provide a better solution. Any help is welcomed :)

**Comment by normen** [comment]
> Yeah, I agree with @normen. `rooms` is supposed to have the full room description. The semi-autoroom feature is an addition to help on the discovery of the IDs, but was never thought to be a long-term solution. Maybe we need to revisit to provide a better solution. Any help is welcomed :)

Simply moving that to a new config option is probably a good idea. Don't remove the current code to use this variable to be backwards compatible and reference the new option in the settings UI.

**Comment by timothy-morph** [comment]
I see :) I'll put my rooms and ids into the room settings to make it stay for now.
Thanks you all for helping!
